.github/workflows/cd.yaml

```diff
@@ -43,7 +43,8 @@ jobs:
           ref: ${{ github.event.pull_request.head.sha }}
       - run: |
           set -x
-          namespace=ci-${{ github.run_number }}
+          runnum=${{ github.run_number }}
+          namespace=ci-temp-${runnum: -1}
```

Comment:
Why do this?

Reply:
The idea is to reduce the number of namespaces used in the zq-test cluster. In this case, we are only using namespaces to get isolation from one test run to another. Having 10 separate namespaces which we cycle through is enough to avoid collisions given that each test run is less than 15 minutes. Namespaces have the potential to leak resources, so I prefer to recycle them rather than have an arbitrary number. The bug we hit yesterday where a CRD (for the Linkerd system I subsequently removed) was unexpectedly interfering with namespace removal is an example of a type of problem we could have with future CRDs, where recycle namespaces would minimize the risk of disruptions.

Reply:
I don't understand how this substitution limits the number of namespaces to 10?

Reply:
@alfred-landrum: `${runnum: -1}` is [Substring Expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html) with a negative index, yielding the last character of `runnum`.

Reply:
Reusing namespaces feels like a bad idea to me precisely because they might contain leaked resources, but I'll defer to you.

Reply:
I do kind of agree with @nwt on not wanting to reuse namespaces.